controller/workitem.go

```diff
@@ -433,7 +433,7 @@ func ConvertJSONAPIToWorkItem(ctx context.Context, appl application.Application,
 			if err != nil {
 				return errors.NewBadParameterError("data.relationships.iteration.data.id", *d.ID)
 			}
-			if _, err = appl.Iterations().Load(ctx, iterationUUID); err != nil {
+			if _, err := appl.Iterations().Exists(ctx, iterationUUID.String()); err != nil {
```

Comment:
Here's an implicit cast from an `InternalError` or a `NotFoundError` to a `BadParameterError` and that is simply not good, I guess. WDYT @hectorj2f ?